restservice/src/main/java/org/itevents/dao/mybatis/sql_session_dao/UserMyBatisDao.java

```diff
@@ -42,7 +44,17 @@ public User getUserByName(String name) {
 
     @Override
     public void addUser(User user, String password) {
-        getSqlSession().insert("org.itevents.dao.mybatis.mapper.UserMapper.addUser", new UserPassword(user, password));
+        try {
+            getSqlSession().insert("org.itevents.dao.mybatis.mapper.UserMapper.addUser", new UserPassword(user, password));
+        } catch (Throwable e) {
+            Throwable t = e;
+            while (t.getCause() != null) {
+                t = t.getCause();
+                if (t instanceof SQLException) {
+                    throw new EntityAlreadyExistsDaoException(t.getMessage(), t);
+                }
+            }
+        }
```

Comment:
@IgorMaksymov I think that this exception cath is wrong because:
1) It is better to catch `Exception`, not `Throwable`; ([1](http://stackoverflow.com/questions/6083248/is-it-a-bad-practice-to-catch-throwable), [2](http://stackoverflow.com/questions/2274102/difference-between-using-throwable-and-exception-in-a-try-catch#));
2) `SQLException` is  throwed not only in constraint violation case but also because of other DB-related reasons;
3) If exception is not `SQLException` it will be swallowed and we can not catch it.